Jack/Property.fs

```diff
@@ -221,28 +343,44 @@ module PropertyBuilder =
         member __.For(m : Property<'a>, k : 'a -> Property<'b>) : Property<'b> =
             Property.bind m k
 
+        member __.For(xs : seq<'a>, k : 'a -> Property<unit>) : Property<unit> =
+            let join m n =
+                Property.bind m (fun _ -> n)
+            let init =
+                Property.success ()
+            xs |> Seq.map k |> Seq.fold join init
+
         member __.Yield(x : 'a) : Property<'a> =
             Property.success x
 
+        member __.Combine(m : Property<unit>, n : Property<'a>) : Property<'a> =
+            Property.bind m (fun _ -> n)
+
+        member __.TryWith(m : Property<'a>, k : exn -> Property<'a>) : Property<'a> =
+            Property.tryWith m k
+
         member __.Bind(m : Gen<'a>, k : 'a -> Property<'b>) : Property<'b> =
             Property.forAll m k
 
         member __.Return(b : bool) : Property<unit> =
             Property.ofBool b
 
+        member __.ReturnFrom(m : Property<'a>) : Property<'a> =
+            m
+
         member __.Delay(f : unit -> Property<'a>) : Property<'a> =
-            Gen.delay (Property.toGen << f) |> Property.ofGen
+            Property.delay f
 
         member __.Zero() : Property<unit> =
-            Property.discard
+            Property.success ()
```

Comment:
Turns out this has to be `success` instead of `discard` otherwise doing effects causes the property to be discarded.

An example of something you can do with this is:
```fs
Property.check <| property {
    // setup
    for n in names do
        try Registry.unregister n with _ -> ()

    // do actual property stuff..
}
```